lld/test/wasm/lto/Inputs/global_var_init1.ll

```diff
@@ -0,0 +1,42 @@
+target datalayout = "e-m:e-p:32:32-p10:8:8-p20:8:8-i64:64-n32:64-S128"
+target triple = "wasm32-unknown-unknown"
+
+; Generated from this C++ code:
+;
+; int foo();
+; inline int unused = foo();
+;
+; int main() {
+;   return foo();
+; }
```

Comment:
I wonder if its possible to reduce these test files something smaller by hand writing them instead?

I'd always prefer something more concise in the test files in possible.

Also, I wonder if we even need to mention `global_var_init` .. presumably this problem applies to all comat groups?  Should we call these something `comdat_ordering.ll`?

Reply:
> I wonder if its possible to reduce these test files something smaller by hand writing them instead?
> 
> I'd always prefer something more concise in the test files in possible.

It was generated from that C++ code but hand-simplified later to delete some unnecessary parts. I think I can delete some more, but the overall structure is still the same. (That `@llvm.global_ctors` is still necessary; without this the error does not occur.) I think I can delete this comment.

> Also, I wonder if we even need to mention `global_var_init` .. presumably this problem applies to all comat groups? Should we call these something `comdat_ordering.ll`?

I don't think this problem is specific to `global_var_init`, but that all bug reports we got contained it kind of makes me think that the condition this bug is triggered is mostly created by `global_var_init`. I think I can rename the tests, but I guess it is fine to leave the variables in the tests as `__cxx_global_var_init`.

Reply:
Hmm, no, turns out I can't really simplify the contents of `@__cxx_global_var_init` much. Even if I try to simplify it a little, the unused contents will be somehow deleted and `@__cxx_global_var_init` itself is deleted. Will rename the tests, and clarify the comment to say it is simplified after generation. (I think it is still better to have what it is generated from)

Reply:
SGTM